relint.py

```diff
@@ -46,17 +52,73 @@ def load_config(path):
             )
 
 
-def lint_file(filename, tests):
-    try:
-        with open(filename) as fs:
-            content = fs.read()
-    except (IsADirectoryError, UnicodeDecodeError):
-        pass
-    else:
+def lint_file(filename, tests, diff=None):
+    if diff and diff.get(filename):
         for test in tests:
-            if any(fnmatch.fnmatch(filename, fp) for fp in test.filename):
+            for line_number, content in diff.get(filename).items():
                 for match in test.pattern.finditer(content):
-                    yield filename, test, match
+                    yield filename, test, match, line_number
+    else:
+        try:
+            with open(filename) as fs:
+                content = fs.read()
+        except (IsADirectoryError, UnicodeDecodeError):
+            pass
+        else:
+            for test in tests:
+                if any(fnmatch.fnmatch(filename, fp) for fp in test.filename):
+                    for match in test.pattern.finditer(content):
+                        yield filename, test, match, None
+
+
+def parse_diff(output):
```

Comment:
where's the doc string?

Reply:
None of your methods have it. :P Just kidding, I'm gonna add it.

Reply:
You don't want to become like me, but better than me don't you ;)